src/main/webapp/app/exercises/quiz/shared/questions/drag-and-drop-question/drag-and-drop-question.component.ts

```diff
@@ -285,8 +300,22 @@ export class DragAndDropQuestionComponent implements OnChanges {
 
     /**
      * counts the amount of right mappings for a question by using the isLocationCorrect Method
+     * drop locations that are correctly left blank are excluded because they are excluded from grading as well
      */
     countCorrectMappings(): void {
         this.correctAnswer = this.question.dropLocations!.filter((dropLocation) => this.isLocationCorrect(dropLocation)).length;
     }
+
+    /**
+     * counts the amount of incorrect mappings for a question by using the isLocationCorrect Method
+     */
+    countIncorrectMappings(): number {
+        return this.question.dropLocations!.filter((dropLocation) => this.isLocationCorrect(dropLocation) === false).length;
+    }
+    /**
+     * counts the amount of drop locations participating in at least one mapping for a question by using the isAssignedLocation Method
+     */
+    countAssignedLocations(): number {
+        return this.question.dropLocations!.filter((dropLocation) => this.isAssignedLocation(dropLocation)).length;
+    }
```

Comment:
as suggested in the comment above (drag-and-drop-question.component.html), I would refactor the code in these methods to something like:

```
export class DragAndDropQuestionComponent implements OnChanges {
    ...
    incorrectLocationMappings: number;
    mappedLocations: number;

    ngOnInit(): void {
        evaluateDropLocations();
    }

    ngOnChanges(): void {
        ...
        evaluateDropLocations();
    }

    evaluateDropLocations(): void {
        // this also avoids the error-prone force unwrapping with (!), maybe this.question.dropLocations?.XYZ would also work
        if (this.question.dropLocations) {
            this.incorrectLocationMappings = this.question.dropLocations!.filter((dropLocation: DropLocation) => this.isLocationCorrect(dropLocation) === false).length;
            this.mappedLocations = this.question.dropLocations!.filter((dropLocation: DropLocation) => this.isAssignedLocation(dropLocation)).length;
        }  
    }
}
```

Reply:
I merged the different methods, introduced the necessary attributes and adapted the documentation.  Have a look again and let me know whether you agree with this modification.